backend/app/models/experiment.py

```diff
@@ -1,42 +1,104 @@
+from __future__ import annotations
+
 from datetime import datetime, timezone
 from functools import partial
 
-from beanie import Document, PydanticObjectId
+import pymongo
+from beanie import Document, Indexed, PydanticObjectId
+from deepdiff import DeepDiff
 from pydantic import Field
 
 from app.models.experiment_template import ExperimentTemplate
 from app.schemas.env_vars import EnvironmentVar
-from app.schemas.states import TemplateState
+from app.schemas.experiment import ExperimentCreate, ExperimentResponse
 
 
 class Experiment(Document):
-    name: str
+    name: Indexed(str, index_type=pymongo.TEXT)  # type: ignore
     description: str
-    publication_ids: list[str]
     updated_at: datetime = Field(default_factory=partial(datetime.now, tz=timezone.utc))
     created_at: datetime = Field(default_factory=partial(datetime.now, tz=timezone.utc))
     created_by: str
+    is_public: bool = False
```

Comment:
I think it would be better to stick with simple standard terms `public` and `archived`.